apis/sgv2-graphqlapi/src/test/java/io/stargate/sgv2/graphql/schema/cqlfirst/dml/fetchers/MutationFetcherTest.java

```diff
@@ -146,4 +167,45 @@ public void mutationAtomicMultipleSelectionWithDifferentOptionsFailTest() {
             "Exception while fetching data (/m1) : options can only de defined once in an @atomic mutation selection",
             "Exception while fetching data (/m2) : options can only de defined once in an @atomic mutation selection");
   }
+
+  @ParameterizedTest
+  @DisplayName("Atomic mutations should be inserted with the correct write timestamp")
+  @MethodSource("provideTTLOptions")
+  public void mutationAtomicMultipleSelectionWithDifferentTTLOptions(String ttl) {
+    String ttlOptionTemplate = "options: { ttl: %s }";
+    String ttlOption = "";
+    if (!Strings.isNullOrEmpty(ttl)) {
+      ttlOption = String.format(ttlOptionTemplate, ttl);
+    }
+
+    String atomicMutationTemplate =
+        "mutation @atomic { "
+            + "m1: insertbooks("
+            + "  value: { title: \"1984\", author: \"G.O.\" },"
+            + "  %s) { applied },"
+            + "m2: insertauthors("
+            + "  value: { author: \"G.O.\", title: \"1984\" },"
+            + "  %s) { applied }"
+            + "}";
+
+    String mutation = String.format(atomicMutationTemplate, ttlOption, ttlOption);
+    ExecutionResult result = executeGraphql(mutation);
+    assertThat(result.getErrors()).isEmpty();
+    Batch batch = getCapturedBatch();
+
+    if (Strings.isNullOrEmpty(ttl)) {
+      assertBookAndAuthorInserts(batch);
+    } else {
+      assertBookAndAuthorInsertsWithTTL(batch, Integer.parseInt(ttl));
+    }
+
+    // Batch timestamp and nowInSeconds parameters should be unset to avoid using them in
+    // BatchHandler's `makeParameters`
+    assertFalse(batch.getParameters().hasTimestamp());
```

Comment:
Might be more idiomatic to use assertThat(batch.getParameters().hasTimestamp()).isFalse() here, given that we're already using assertJ